src/science/mathematics/statistics/central-tendency.md

```diff
@@ -0,0 +1,88 @@
+### Measures of central tendency
+
+[![std-badge]][std]
+
+Measures of central tendency summarize a set of data as an "average" or "expected" value that is likely to be representative of a typical member of the data set in some way. The most common measures of central tendency are the
+- mean (the sum of all measurements divided by the number of measurements in the set)
+- median (the value in the middle of a sorted set, or mean of the two middle values when the size of the set is even), and
+- mode (element of the set that appears with the highest frequency)
+
+This example calculates these measures for a data set contained within a Rust array, passed to functions as a [`slice`]. Any collection that can produce an Iterator, such as a Vector, may use a similar approach.
+
+The example calculates the mean by producing an iterator of references over the data, and using [`sum`] to determine the total value. The slice itself contains the number of elements, obtained via [`len`]. A simple division provides the mean, with conversions to floating point types to account for rational components of the result.
+
+```rust
+fn main() {
+    let data = [3, 1, 6, 1, 5, 8, 1, 8, 10, 11];
+
+    let sum = data.iter().sum::<i32>() as f32;
+    let count = data.len();
+
+    let mean = match count {
+       positive if positive > 0 => Some(sum  / count as f32),
+       _ => None
+    };
+
+    println!("Mean of the data is {:?}", mean);
+}
+```
+
+Calculating the median [`sort`]s the data in-place, requiring the input slice to be mutable. If the number of elements is even, calculate the mean of the middle two measurements, which can be passed to the previous mean function as a sub[`slice`] of the original slice. For odd-sized data sets, simply pick the measurement from the middle of the sorted set.
```

Comment:
Shouldn't the median not affect the data set?  Is there a way to calculate the median without mutating the data?

Reply:
There are a few ways, though a straight `clone` and `sort` isn't very performant. I might be able to conjure a good implementation of the quickselect algorithm, though that may be slightly complex for one of these examples (while it could show off some of Rust's finer points). Bears some thinking about...